packages/consensus/src/types/utreexo.cairo

```diff
@@ -46,6 +47,22 @@ pub struct UtreexoState {
     pub num_leaves: u64,
 }
 
+#[generate_trait]
+pub impl UtreexoStateImpl of UtreexoStateTrait {
+    fn validate_and_apply(ref self: UtreexoState, mut utxo_set: UtxoSet) -> Result<(), ByteArray> {
+        if utxo_set.num_cached != 0 {
```

Comment:
I would probably move that check to the `UtxoSet` (maybe a separate method) to clearly distinct between accumulator updates and double spending prevention

Reply:
I've moved it to a separate method `UtxoSet.finalize()`, where also added additional checks for better consistency (they are commented out at the moment, until utreexo is enabled). Please, check it out.